### PR TITLE
Update Knative dependencies to v1.1

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

- :broom: Update Knative dependencies to v1.1

/cc @lance

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,37 +11,39 @@ require (
...
+	github.com/docker/docker v20.10.12+incompatible
 	github.com/docker/docker-credential-helpers v0.6.4
 	github.com/docker/go-connections v0.4.0
 	github.com/go-git/go-billy/v5 v5.3.1
 	github.com/go-git/go-git/v5 v5.4.2
-	github.com/google/go-cmp v0.5.6
-	github.com/google/go-containerregistry v0.6.0
+	github.com/go-logr/logr v1.2.2 // indirect
+	github.com/google/go-cmp v0.5.7
+	github.com/google/go-containerregistry v0.8.1-0.20220219142810-1571d7fdc46e
```
Comment (dsimansk) [inline go.mod:22]: I'll remove this bit for now.

File: docker/pusher.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type PusherDockerClient interface {
 	ImageLoad(context.Context, io.Reader, bool) (types.ImageLoadResponse, error)
 	ImageTag(context.Context, string, string) error
 	ImagePush(ctx context.Context, ref string, options types.ImagePushOptions) (io.ReadCloser, error)
+	ImageInspectWithRaw(context.Context, string) (types.ImageInspect, []byte, error)
```
Comment (lance) [inline docker/pusher.go:43]: Sorry for my ignorance. Is this just needed because of the bump in the docker dependency version?

Comment (matejvasek) [reply docker/pusher.go:43]: Probably dep to go-containers lib.

File: vendor/github.com/google/go-containerregistry/pkg/v1/daemon/options.go [hunk 3]
```diff
@@ -99,4 +99,5 @@ type Client interface {
 	ImageSave(context.Context, []string) (io.ReadCloser, error)
 	ImageLoad(context.Context, io.Reader, bool) (types.ImageLoadResponse, error)
 	ImageTag(context.Context, string, string) error
+	ImageInspectWithRaw(context.Context, string) (types.ImageInspect, []byte, error)
```
Comment (matejvasek) [inline vendor/github.com/google/go-containerregistry/pkg/v1/daemon/options.go:102]: Here @lance .

File: docker/pusher.go (hunk 2 above)
Comment (dsimansk) [reply docker/pusher.go:43]: Yep, that's an updated interface from go-containers. Should I refactor it to Matej's suggestion? It's definitely cleaner.

**Pull request merged**